lib/mrb/scripts/command_line/grndb.rb

```diff
@@ -402,6 +406,42 @@ def failed_to_open(name)
           failed(message)
         end
       end
+
+      class Recoverer
+        attr_writer :database
+        attr_writer :force_truncate
+
+        def initialize
+          @context = Context.instance
+        end
+
+        def recover
+          @database.recover
+          if @force_truncate
+            truncate_corrupt_objects
+          end
+        end
+
+        def truncate_corrupt_objects
+          @database.each do |object|
+            next unless object.corrupt?
+            logger = @context.logger
+            object_path = object.path
+            Dir.foreach(File.dirname(object_path)) do |path|
+              if path.start_with?("#{File.basename(object_path)}.")
+                begin
+                  File.unlink("#{File.dirname(object_path)}/#{path}")
+                  message = "Removed file which is related to corrupted #{object_path}: #{path}"
```

Comment:
Variable in message should be surrounded with `<...>`. You can find useful information such as empty value and garbage spaces at the last by this technique.